omod/src/main/java/org/openmrs/module/ordertemplates/web/resource/OrderTemplatesResource.java

```diff
@@ -196,6 +197,35 @@ protected PageableResult doSearch(RequestContext requestContext) {
 		OrderTemplateCriteriaBuilder builder = new OrderTemplateCriteriaBuilder();
 		builder.setDrug(drug).setConcept(concept);
 		List<OrderTemplate> orderTemplates = getService().getOrderTemplateByCriteria(builder.build());
-		return new NeedsPaging(orderTemplates, requestContext);
+		return new NeedsPaging<>(orderTemplates, requestContext);
+	}
+	
+	@Override
+	public Object getProperty(OrderTemplate instance, String propertyName) throws ConversionException {
+		if ("drug".equals(propertyName)) {
+			return instance.getDrug();
+		}
+		return super.getProperty(instance, propertyName);
+	}
+	
+	@Override
+	public void setProperty(Object instance, String propertyName, Object value) throws ConversionException {
```

Comment:
The test called `testCreateOrderTemplate` was failing to convert drug UUIDs to Drug objects when creating order templates. I have added this custom property handling in OrderTemplatesResource to properly convert drug UUIDs to Drug objects.